Implemented Avl_automation Enhancements

Filipe-Cavalheiro opened the pull request:
Two new tools were implemented and one was updated

# Summary
This pull request introduces two new tools and an update to Avl_automation, aiming to improve simulation accuracy and enhance aerodynamic modeling.

## New Tool: QGC2ROMFS_parameters
 - Ensures that parameters set in QGroundControl (QGC) are correctly transferred to Gazebo.
 - Addresses a critical issue where discrepancies between QGC and Gazebo parameters caused severe crashes during real-world tests.
 - Can be executed manually or with a single-line command, as described in the README.

## New Tool: ROMFS2QGC_parameters (In Development)
 - Enables the export of simulation-tuned parameters back to QGC for real-world use.
 - Provides an automated solution to ensure consistency between simulation and physical flight behavior.
 
## Updated Tool: Avl_automation
 - Now supports multiple control surfaces per wing surface.
 - A new example has been added to demonstrate this feature.

## Feedback & Next Steps
Any feedback is appreciated to ensure a smooth integration of these improvements. Let me know if anything needs adjustment!

Jaeyoung-Lim reviewed (changes requested):
Thank you for the contribution!

- Could you separate the two features to two different PRs?

- I am not sure if I understand correctly the QGC2ROMFS feature. Wouldnt this be agnostic to the simulator used, and simply a separate tool for px4? Why is this useful for gazebo?

Filipe-Cavalheiro commented:
Hi @Jaeyoung-Lim,

Thank you for your reply.

I’ve only used Gazebo Classic, so I hadn’t considered the versatility of QGC2ROMFS across different simulators. As long as the parameter definition format remains consistent across all simulators, then yes! Both QGC2ROMFS and ROMFS2QGC will have a PR submitted to PX4, with only the updated AVL_automation remaining specific to Gazebo.

Jaeyoung-Lim commented on tools/.vscode/c_cpp_properties.json:
```diff
@@ -0,0 +1,20 @@
+    }
+  ],
+  "version": 4
+}
```
Please remove vscode artifacts from the PR

Jaeyoung-Lim commented:
@Filipe-Cavalheiro There seems to be vscode artifacts and pycache submitted as part of the PR. After these are removed, I can review the PR

Filipe-Cavalheiro commented:
Hi @Jaeyoung-Lim,
The artifacts have been removed, apologies for the newbie mistake, and both QGC2ROMFS and ROMFS2QGC have been transferred to new PRs in the PX4 repository:

- QGC2ROMFS: [New Tool: QGC2ROMFS](https://github.com/PX4/PX4-Autopilot/pull/24373)
- ROMFS2QGC: [New Tool: ROMFS2QGC (In Development)](https://github.com/PX4/PX4-Autopilot/pull/24374)

To be completely honest, I do believe that the AVL automation with multiple control surfaces per wing is functional. However, I haven’t been able to create a YAML file for AVL that I’m fully satisfied with. This is mainly because aerodynamics are outside my area of expertise so I believe that my definition of the wings (easy_glider4) is not totally correct.

hamishwillee commented:
Assuming this changes the way the AVL tool is run, would be great if we could have a docs update after this merges: https://docs.px4.io/main/en/sim_gazebo_gz/tools_avl_automation.html#advanced-lift-drag-avl-automation-tool

Filipe-Cavalheiro commented:
Hi @hamishwillee,
I've updated the [README](https://github.com/PX4/PX4-gazebo-models/pull/79/commits/d1356564e94f4b504851fd144d082d43511084d1) to reflect the new usage of the AVL automation. No changes were made directly to AVL—only to the parsers: from YAML to AVL (input_parser) and from AVL output to Gazebo (avl_output).

I have finally also discovered what was wrong with the parser and fixed it, the easy glider 4 now works as expected, further details located in the commit [Fix aileron inversion logic in AVL output parser](https://github.com/PX4/PX4-gazebo-models/pull/79/commits/7132ff87ad1f0008700b3c16437eb19c4f60c021). 

As always, any changes needed to get this push through are more than welcome.

hamishwillee commented:
Thanks. I'm on holiday. If no one else gets to this, I will check it out in 2 weeks.

Jaeyoung-Lim commented on tools/avl_automation/avl_out_parse_1_0.py:
```diff
@@ -0,0 +1,335 @@
+#!/usr/bin/env
+
```
What is the difference with this and the original avl_out_parse_0_0.py? Shouldn't they be one file?

Filipe-Cavalheiro replied:
The main difference between this and the original avl_out_parse_0_0.py is that the new version includes some variable name changes and updates to how the received data is parsed for Gazebo. Specifically, [this commit](https://github.com/PX4/PX4-gazebo-models/pull/79/commits/7132ff87ad1f0008700b3c16437eb19c4f60c021) corrects the inversion order of the ailerons in the simulation, which was necessary for proper behavior in Gazebo.

If it's deemed better for clarity and maintenance, we can remove avl_out_parse_0_0.py entirely and just use the updated version going forward.

Jaeyoung-Lim replied:
The problem is that it is not even clear what have changed between the two PRs making it hard to review. If you think it is an improvement please change the file.

Jaeyoung-Lim commented on tools/avl_automation/input_avl_0_0.py:
```diff
@@ -297,7 +297,7 @@ def main():
 
 	# Call main function of avl parse script to parse the generated AVL files.
-	avl_out_parse.main(plane_name,frame_type,AR,mac,ref_pt_x,ref_pt_y,ref_pt_z,num_ctrl_surfaces,area,ctrl_surface_order,inputs.avl_path)
+	avl_out_parse_0_0.main(plane_name,frame_type,AR,mac,ref_pt_x,ref_pt_y,ref_pt_z,num_ctrl_surfaces,area,ctrl_surface_order,inputs.avl_path)
```
why was this file renamed?

Filipe-Cavalheiro replied:
Since there are two versions of the parser, the original input parser currently relies on the original output parser. However, this setup no longer makes sense—the updated output parser should always be used instead. Therefore, the deprecated avl_out_parse_0_0.py and input_avl_parse_0_0.py will be removed to avoid redundancy and ensure consistency.
